### PR TITLE
[#515] Remove all Rx things

### LINKED ISSUE #515: Remove RxSwift and its libraries for UIKit variant
## Why

We currently supports iOS 14 as the minimum version, so it's better to remove RxSwift and its libraries (e.g., RxAlamofire, RxNimble, etc.) out of the templates.
 
## Who Benefits?

iOS developers

### PR DESCRIPTION
**Pull request by nkhanh44**
- Close #515

## What happened 👀

- Removed `RxNimble`, `RxSwift`, `RxDataSources`, `RxCocoa`, and `RxAlamofire`.
- Added `Alamofire`, `JSONAPIMapper`.
- Updated `NetworkAPIProtocol`, and `NetworkAPI` to the Combine version.

## Insight 📝

- N/a

## Proof Of Work 📹

- No more `Rx` search found
- Test SwiftUI and UIKit Install Script / Test (push) are all successful

### CODE REVIEW

**Review by blyscuit — APPROVED** [review]

File: Tuist/Interfaces/UIKit/Sources/Data/NetworkAPI/Core/NetworkAPIProtocol.swift [hunk 1]
```diff
@@ -3,12 +3,14 @@
 //
 
 import Alamofire
-import RxAlamofire
-import RxSwift
+import Combine
```
Comment (suho) [inline Tuist/Interfaces/UIKit/Sources/Data/NetworkAPI/Core/NetworkAPIProtocol.swift:6]: @nkhanh44 Could you check how much effort you need to implement `async/await` for `NetworkAPI`, just want to minimize the framework in this layer

**Review by Thieurom — APPROVED** [review]

File: Tuist/Interfaces/UIKit/Sources/Data/NetworkAPI/Core/NetworkAPIProtocol.swift (hunk 1 above)
Comment (nkhanh44) [reply Tuist/Interfaces/UIKit/Sources/Data/NetworkAPI/Core/NetworkAPIProtocol.swift]: @suho I updated it here https://github.com/nimblehq/ios-templates/pull/525/commits/670ef6bf55b77120df4fe930d6b108e4edaf17d3 🙏  Please help to review again

**Review by blyscuit — COMMENTED** [review]
Looks good, not sure if we should update SwiftUI with Async too 🤔

File: Tuist/Interfaces/UIKit/Sources/Data/NetworkAPI/Core/NetworkAPIProtocol.swift [hunk 2]
```diff
@@ -17,28 +18,29 @@ extension NetworkAPIProtocol {
...
-        .responseData()
-        .flatMap { _, data -> Observable<T> in
-            Observable.create { observer in
-                do {
-                    let decodable = try decoder.decode(T.self, from: data)
-                    observer.on(.next(decodable))
-                } catch {
-                    observer.on(.error(error))
+    ) async throws -> T {
+        try await withCheckedThrowingContinuation { continuation in
```
Comment (suho) [inline Tuist/Interfaces/UIKit/Sources/Data/NetworkAPI/Core/NetworkAPIProtocol.swift:22]: Not sure did you check about this yet: https://github.com/Alamofire/Alamofire/blob/master/Documentation/AdvancedUsage.md#using-alamofire-with-swift-concurrency

We can use async await directly

**Comment by suho** [comment]
@blyscuit For Tuist/Interfaces, we should focus on Presentation Layer only, Data and Domain could be shared

**Comment by blyscuit** [comment]
@suho @nkhanh44 I meant [this files](https://github.com/nimblehq/ios-templates/blob/main/Tuist/Interfaces/SwiftUI/Sources/Data/NetworkAPI/NetworkAPI.swift) and the children 🙏

**Comment by suho** [comment]
@blyscuit Yes, we can move it back to `Project/Sources/Data`

File: Tuist/Interfaces/UIKit/Sources/Data/NetworkAPI/Core/NetworkAPIProtocol.swift (hunk 2 above)
Comment (nkhanh44) [reply Tuist/Interfaces/UIKit/Sources/Data/NetworkAPI/Core/NetworkAPIProtocol.swift]: @suho Good point, I updated to use them directly at https://github.com/nimblehq/ios-templates/pull/525/commits/034edaea3ac6fdc2e0990fa29d7a62a9622525a4, also move that `Data` folder to shared  `Project/Sources/Data`🙏  cc: @blyscuit

**Review by blyscuit — APPROVED** [review]

**Review by suho — APPROVED** [review]

**Pull request merged**